### PR TITLE
macOS wheel perf check

### PR DESCRIPTION
**Pull request by antalszava**
**Context**
We'd like to check the performance of PennyLane-Lightning on macOS. For this we use the wheel built on macOS and the PennyLane benchmarking tool (found in the `benchmark` folder of PennyLane).

**Results**
Performance comparison of `default.qubit` and `lightning.qubit` for the 'Entangling layers' benchmark run on macOS:

[With 2 wires](https://github.com/PennyLaneAI/pennylane-lightning/pull/38/checks?check_run_id=981841206#step:6:953):
```
Timing: 'Entangling layers' benchmark on default.qubit, QNode
10 loops, 5 runs
Timing per loop: [0.07821284 0.07821478 0.07785126 0.07677576 0.07520293]
Timing: 'Entangling layers' benchmark on lightning.qubit, QNode
10 loops, 5 runs
Timing per loop: [0.04033666 0.039648   0.04055351 0.03933717 0.03958955]
```
[With 6 wires](https://github.com/PennyLaneAI/pennylane-lightning/pull/38/checks?check_run_id=981841206#step:6:988):
```
Timing: 'Entangling layers' benchmark on default.qubit, QNode
10 loops, 5 runs
Timing per loop: [0.4530408  0.45083251 0.45257708 0.45071833 0.45597058]
Timing: 'Entangling layers' benchmark on lightning.qubit, QNode
10 loops, 5 runs
Timing per loop: [0.18767837 0.18495741 0.18629023 0.18913664 0.18902163]
```
[With 10 wires](https://github.com/PennyLaneAI/pennylane-lightning/pull/38/checks?check_run_id=981841206#step:6:1023):
```
Timing: 'Entangling layers' benchmark on default.qubit, QNode
10 loops, 5 runs
Timing per loop: [1.81512918 1.81848905 1.79244642 1.79102871 1.82041465]
Timing: 'Entangling layers' benchmark on lightning.qubit, QNode
10 loops, 5 runs
Timing per loop: [1.02808973 1.02623959 1.04195447 1.03139232 1.03201263]
```

### CODE REVIEW

File: .github/workflows/performance_checks.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  CIBW_TEST_COMMAND: |
+    git clone https://github.com/PennyLaneAI/pennylane.git
+    cd pennylane && pip install -e .
+    pytest pennylane/devices/tests --device=lightning.qubit --skip-ops -x --tb=short --no-flaky-report
+
+    # Benchmark performance
+    python3 benchmark/benchmark.py -d default.qubit,lightning.qubit time bm_entangling_layers -w 2
+    python3 benchmark/benchmark.py -d default.qubit,lightning.qubit time bm_entangling_layers -w 6
+    python3 benchmark/benchmark.py -d default.qubit,lightning.qubit time bm_entangling_layers -w 10
```
Comment (antalszava) [inline .github/workflows/performance_checks.yml:53]: Added these for running the benchmarks

**Comment by trbromley** [comment]
@antalszava also what about this one?

**Comment by trbromley** [comment]
Maybe this could be closed and revisited later if needed?

**Comment by antalszava** [comment]
This was a quick way of checking the performance of the built wheels at the time when was drafted. As we'll go with a more sophisticated benchmarking approach, I'll close this for now (could be reopened later if we'd like to revisit).

**Pull request closed without merging**